src/hooks/useHandleSelectorComponent.tsx

```diff
@@ -1,12 +1,26 @@
-import { useState } from 'react'
+import { SelectorType } from '@/components/inputs/Selector'
+import { setCreateTaskFields } from '@/redux/features/createTaskSlice'
+import store from '@/redux/store'
+import { WorkflowStateResponse } from '@/types/dto/workflowStates.dto'
+import { useEffect, useState } from 'react'
 
-export const useHandleSelectorComponent = ({ item }: { item: unknown }) => {
+export const useHandleSelectorComponent = ({ item, type }: { item: unknown; type: SelectorType }) => {
   const [renderingItem, setRenderingItem] = useState<unknown>(item)
 
   const updateRenderingItem = (newValue: unknown) => {
     setRenderingItem(newValue)
   }
 
+  useEffect(() => {
+    if (type === SelectorType.STATUS_SELECTOR) {
+      store.dispatch(setCreateTaskFields({ targetField: 'workflowStateId', value: (item as WorkflowStateResponse).id }))
+    }
+
+    if (type === SelectorType.ASSIGNEE_SELECTOR) {
+      //do this later while integrating assignee
+    }
+  }, [type])
```

Comment:
Either add `item` to dependency array, or disable lint warning for this line. Lets not do a global warning disable for `react-hooks/exhaustive-deps` just yet though

Reply:
Added another dependency `item` in the array